src/framework/preferences/js/LetterSpaceEnactor.js

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2017 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_3_0_0 = fluid_3_0_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    /*******************************************************************************
+     * letterSpace
+     *
+     * Sets the letter space on the container to the number of units to increase
+     * the letter space by. If a negative number is provided, the space between
+     * characters will decrease. Setting the size to 0 will use the default letter
+     * space.``
+     *******************************************************************************/
+
+    // Note that the implementors need to provide the container for this view component
+    fluid.defaults("fluid.prefs.enactor.letterSpace", {
+        gradeNames: ["fluid.prefs.enactor", "fluid.viewComponent"],
+        preferenceMap: {
+            "fluid.prefs.letterSpace": {
+                "model.value": "default"
+            }
+        },
+        fontSizeMap: {},  // must be supplied by implementors
+        members: {
+            root: {
+                expander: {
+                    "this": "{that}.container",
+                    "method": "closest", // ensure that the correct document is being used. i.e. in an iframe
+                    "args": ["html"]
+                }
+            }
+        },
+        invokers: {
+            set: {
+                funcName: "fluid.prefs.enactor.letterSpace.set",
+                args: ["{that}", "{arguments}.0"]
+            }
+        },
+        modelListeners: {
+            unit: {
+                listener: "{that}.set",
+                args: ["{change}.value"]
+            }
+        },
+        modelRelay: {
+            target: "unit",
+            singleTransform: {
+                type: "fluid.transforms.round",
+                scale: 1,
+                input: {
+                    transform: {
+                        "type": "fluid.transforms.linearScale",
+                        "offset": -1,
+                        "input": "{that}.model.value"
+                    }
+                }
+            }
+
+        }
+    });
+
+    fluid.prefs.enactor.letterSpace.set = function (that, units) {
+        var targetSize = units  ? units + "em" : "normal";
```

Comment:
What if `letter-spacing` css has already been set on the html? Shall this function fetch it and performs an multiplication on the original value instead of a straight overwrite?

Reply:
@cindyli and spoke about this today after I had performed some experimentation. So the situation is a bit trickier for letter spacing. There are a couple of factors 1) we don't really know the base spacing (kerning, tracking) used within a font. The letter-spacing css property actually just adds space to whatever this original value is. Meaning we don't ever do true multiplication like in font size adjustments. 2) If an original CSS value was set to say 0.2em, and user increases the letter-spacing preference in UIO by 0.5. If we do a multiplication of these values, the size would actually become 0.1em, making the space smaller. Due to all of these factors we decided to add the original value to the UIO adjustment. That is is 0.2 + 0.5 resulting in 0.7em value being set.